lua/entities/gmod_wire_ranger.lua

```diff
@@ -119,7 +122,9 @@ function ENT:Think()
 
 	local tracedata = {}
 	tracedata.start = self:GetPos()
-	if (self.Inputs.X.Value == 0 and self.Inputs.Y.Value == 0) then
+	if self.Inputs.Target.Value ~= Vector() then
+		tracedata.endpos = self:GetPos()+(self:GetTarget()-self:GetPos()):Angle():Forward()*self:GetBeamLength()
```

Comment:
Change `:Angle():Forward()` to `:GetNormalized()` and on the line after it, `if tracedata.endpos[1] ~= tracedata.endpos[1] then tracedata.endpos = Vector(self:GetBeamLength(), 0, 0) end`